### PR TITLE
M2-1611 Chart tooltip

### PR DESCRIPTION
**Pull request by wildkiomi**

### CODE REVIEW

**Comment by IhorNesterenko99** [comment]
I think you need to add a pull request also to to the development branch.

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Charts/ScatterChart/ChartTooltip/ChartTooltip.types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export type ChartTooltipPosition = {
+  top: string;
+  left: string;
+};
+
+export type ChartTooltipProps = {
+  position: ChartTooltipPosition;
+  data: any;
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Charts/ScatterChart/ChartTooltip/ChartTooltip.types.ts:8]: pls don't use any

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Charts/ScatterChart/ScatterChart.const.ts [hunk 2]
```diff
@@ -20,7 +20,11 @@ const commonConfig = {
   },
 };
 
-export const getOptions = (lang: keyof typeof locales, data: Data) => {
+export const getOptions = (
+  lang: keyof typeof locales,
+  data: Data,
+  tooltipHandler: (context: any) => void,
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Charts/ScatterChart/ScatterChart.const.ts:26]: pls don't use any

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Charts/ScatterChart/ScatterChart.tsx [hunk 3]
```diff
@@ -16,20 +19,50 @@ import { locales } from 'shared/consts';
...
 
 export const ScatterChart = ({ height = '50px' }: ScatterChartProps) => {
   const { i18n } = useTranslation('app');
+  const [tooltipPosition, setTooltipPosition] = useState<ChartTooltipPosition | null>(null);
+  const [tooltipData, setTooltipData] = useState<any>(null);
+  const chartRef = useRef<ChartJSOrUndefined<'scatter', { x: Date; y: number }[], unknown> | null>(
+    null,
+  );
+
+  const tooltipHandler = (context: any) => {
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Charts/ScatterChart/ScatterChart.tsx:35]: don't use any

**Review by muhurov — APPROVED** [review]

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**